File: problems/Find First and Last Position of Element in Sorted Array/searchRange.cs
public class Solution {
    public int[] SearchRange(int[] nums, int target) {
        var left = getIndex(nums, target);

        if (left == nums.Length || target != nums[left]) {
            return new int[2] {-1, -1};
        }

        var right = getIndex(nums, target, false);

        return new int[2] {left, right - 1};
    }

    private int getIndex(int[] nums, int target, bool left = true) {
        var l = 0;
        var r = nums.Length;

        while (l < r) {
            var mid = l + ((r - l) >> 1);

            if (nums[mid] > target || (left && nums[mid] == target)) {
                r = mid;
            } else {
                l = 1 + mid;
            }
        }

        return l;
    }
}

File: problems/Find First and Last Position of Element in Sorted Array/searchRange.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int leftIdx = binarySearch(nums, target, true);

        if (0 > leftIdx) {
            return new int[] {-1, -1};
        }

        int rightIdx = binarySearch(nums, target, false);

        return new int[] {leftIdx, rightIdx};
    }

    private int binarySearch(int[] nums, int target, boolean isLeft) {
        int left = 0;
        int right = nums.length;

        while (1 + left < right) {
            int mid = left + ((right - left) / 2);

            if (nums[mid] == target) {
                if (isLeft) {
                    right = mid;
                } else {
                    left = mid;
                }
            } else if (nums[mid] < target) {
                left = mid;
            } else {
                right = mid;
            }
        }

        if (nums.length > left && nums[left] == target) {
            return left;
        } else if (nums.length > right && nums[right] == target) {
            return right;
        }

        return -1;
    }
}
